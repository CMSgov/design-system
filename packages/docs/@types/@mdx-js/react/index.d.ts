declare module '@mdx-js/react' {
  const MDXProvider: React.ComponentType;
  export { MDXProvider };
}
