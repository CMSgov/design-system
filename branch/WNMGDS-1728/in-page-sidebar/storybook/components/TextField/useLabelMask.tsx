import React, { useState, useRef } from 'react';
import classNames from 'classnames';
import uniqueId from 'lodash/uniqueId';
import { TextInputProps } from './TextInput';

export type MaskFunction = (rawInput: string, valueOnly?: boolean) => string;

export const RE_DATE = /^(\d{1,2})[\D]?(\d{1,2})?[\D]?(\d{1,4})?/;

/**
 * Takes the string value from an input and returns a string
 * with appropriate date format masking applied concatenated
 * with the date hint text, MM/DD/YYYY.
 *
 * `valueOnly` defaults to false. If true, returns formatted
 * string without additional hint text.
 */
export const DATE_MASK: MaskFunction = (rawInput = '', valueOnly = false) => {
  const match = RE_DATE.exec(rawInput);
  let formattedDate = '';
  if (match) {
    const [month, day, year] = match.slice(1);
    formattedDate = [
      // We treat all non-numeric characters as a delimiter. If they're using a
      // delimiter after a month or day, we interpret that as the user supplying
      // a single digit for month or day, which we will automatically pad for them.
      month && month.padStart(2, '0'),
      day && day.padStart(2, '0'),
      year,
    ]
      .filter((s) => s)
      .join('/');
  }
  if (valueOnly) {
    return formattedDate;
  }
  const hint = 'MM/DD/YYYY';
  const hintSub = hint.substring(formattedDate.length);
  return formattedDate + hintSub;
};

export function useLabelMask(maskFn: MaskFunction, originalInputProps: TextInputProps) {
  // TODO: Once we're on React 18, we can use the `useId` hook
  const labelMaskId = useRef(uniqueId('labelmask_')).current;
  const [focused, setFocused] = useState(false);
  const { onFocus, onBlur, onChange } = originalInputProps;
  const value = originalInputProps.value?.toString() ?? '';

  const inputProps = {
    ...originalInputProps,
    defaultValue: undefined,
    onFocus: (e) => {
      if (onFocus) {
        onFocus(e);
      }

      setFocused(true);
    },
    onBlur: (e) => {
      const maskedValue = maskFn(value.toString(), true);
      e.currentTarget.value = maskedValue;
      e.target.value = maskedValue;

      if (onChange) {
        (onChange as any)(e);
      }

      if (onBlur) {
        (onBlur as any)(e);
      }

      setFocused(false);
    },
    type: 'text',
    inputMode: 'numeric' as const,
    'aria-describedby': classNames(originalInputProps['aria-describedby'], labelMaskId),
  };

  return {
    labelMask: (
      <div className="ds-c-label-mask" id={labelMaskId}>
        <span className={classNames(focused && 'ds-u-visibility--screen-reader')}>
          {maskFn('')}
        </span>
        <span className={classNames(!focused && 'ds-u-display--none')} aria-hidden="true">
          {maskFn(value)}
        </span>
      </div>
    ),
    inputProps,
  };
}

export default useLabelMask;
