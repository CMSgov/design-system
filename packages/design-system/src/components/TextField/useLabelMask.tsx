import React from 'react';
import { useEffect, useState } from 'react';
import classNames from 'classnames';
import { TextInputProps } from './TextInput';
import mergeIds from '../utilities/mergeIds';
import useId from '../utilities/useId';

export type MaskFunction = (rawInput: string, valueOnly?: boolean) => string;

/**
 * Regular expressions for matching commonly used masks, ensure capture groups are
 * set for each set of items that are separated by a delimiter, delimiters are
 * optional and so should be shown as well with the `?` optional flag.
 */
export const RE_DATE = /^(\d{1,2})[\D]?(\d{1,2})?[\D]?(\d{1,4})?/;
export const RE_PHONE = /^\(?(\d{1,3})?\)?[\s.-]?(\d{1,3})?[\s.-]?(\d{1,4})?/;
export const RE_SSN = /([*\d]{1,3})[\s.-]?([*\d]{1,2})?[\s.-]?([\d{1,4}]+)?/;
export const RE_ZIP = /(\d{1,5})/;
export const RE_CURRENCY = /\$?[\d,.-]*/;

/**
 * This function returns a mask function which returns either the formatted match only, or
 * the formatted match + the hint substring which the regular expressions above match against
 * when determining what content has been filled in.
 */
const makeMask = (regex: RegExp, hint: string, formatter: (stringMatch: string[]) => string) => {
  return (rawInput = '', valueOnly = false) => {
    const match = regex.exec(rawInput);
    const formattedMatch = match ? formatter(match) : '';

    if (valueOnly) {
      return formattedMatch;
    }

    const hintSub = hint.substring(formattedMatch.length);
    return formattedMatch + hintSub;
  };
};

/**
 * The date mask automatically pads months and days that are one digit
 */
export const DATE_MASK: MaskFunction = makeMask(RE_DATE, 'MM/DD/YYYY', (match) => {
  const [month, day, year] = match.slice(1);
  const formattedDate = [
    // We treat all non-numeric characters as a delimiter. If they're using a
    // delimiter after a month or day, we interpret that as the user supplying
    // a single digit for month or day, which we will automatically pad for them.
    month && month.padStart(2, '0'),
    day && day.padStart(2, '0'),
    year,
  ]
    .filter((s) => s)
    .join('/');

  return formattedDate;
});

/**
 * Formatting for US phone numbers
 */
export const PHONE_MASK: MaskFunction = makeMask(RE_PHONE, '###-###-####', (match) => {
  return match
    .slice(1)
    .filter((s) => s)
    .join('-');
});

/**
 * Formatting for US Postal codes, this could be expanded to allow for 9 digit numbers
 */
export const ZIP_MASK: MaskFunction = makeMask(RE_ZIP, '#####', (match) => {
  return match[1];
});

/**
 * Formatting for social security numbers.
 */
export const SSN_MASK: MaskFunction = makeMask(RE_SSN, '###-##-####', (match) => {
  /**
   * for future notice, there are rules regarding social security numbers, these could be validated
   * if we were to add error handling to useLabelMask as an additional set of functionality
   * The first part should have 3 digits and should not be 000, 666, or between 900 and 999.
   * The second part should have 2 digits and it should be from 01 to 99.
   * The third part should have 4 digits and it should be from 0001 to 9999.
   */
  return match
    .slice(1)
    .filter((s) => s)
    .join('-');
});

/**
 * Does the same thing as SSN_MASK except that it obfuscates the first five digits
 */
export const SSN_MASK_OBFUSCATED: MaskFunction = (rawInput: string, valueOnly?: boolean) => {
  // Use the normal SSN_MASK function just to clean the raw input and format it
  const formatted = SSN_MASK(rawInput, true);
  // We only hide the first five digits of the SSNs
  const obfuscation = '***-**';

  let obfuscated: string;
  if (formatted.length < obfuscation.length) {
    obfuscated = obfuscation.substring(0, formatted.length);
  } else {
    obfuscated = obfuscation + formatted.substring(obfuscation.length);
  }

  return SSN_MASK(obfuscated, valueOnly);
};

/**
 * Currency mask is a little different, we need to modify the incoming content to strip
 * out any commas or dollar signs before evaluating it via the Intl.NumberFormat function.
 */
export const CURRENCY_MASK = makeMask(RE_CURRENCY, '$', (match) => {
  const signed = match[0].includes('-');
  const stripped = match[0].replace(/[^0-9.]/g, '');
  const clipped = stripped.includes('.') ? stripped.slice(0, stripped.indexOf('.') + 3) : stripped;
  const USDollar = new Intl.NumberFormat('en-US', { style: 'currency', currency: 'USD' });
  const formatted = USDollar.format(Number(clipped)).replace(/\.00/, '');

  if (Number(clipped) > 0) {
    return signed ? '-' + formatted : formatted;
  } else {
    return '';
  }
});

export function useLabelMask(maskFn: MaskFunction, originalInputProps: TextInputProps) {
  // TODO: Once we're on React 18, we can use the `useId` hook
  const generatedId = useId('label-mask--');
  const labelMaskId = originalInputProps.id ? `${originalInputProps.id}__label-mask` : generatedId;
  const [focused, setFocused] = useState(false);
  const { onFocus, onBlur, onChange } = originalInputProps;
  const value =
    originalInputProps.value?.toString() ?? originalInputProps.defaultValue?.toString() ?? '';
  const [currentValue, setCurrentValue] = useState(value);

  useEffect(() => {
    setCurrentValue(value);
  }, [value, setCurrentValue]);

  const inputProps = {
    ...originalInputProps,
    defaultValue: undefined,
    value: currentValue,
    onChange: (e: React.ChangeEvent<HTMLInputElement>) => {
      setCurrentValue(e.currentTarget.value);

      if (onChange) {
        onChange(e);
      }
    },
    onFocus: (e: React.FocusEvent<HTMLInputElement>) => {
      if (onFocus) {
        onFocus(e);
      }

      setFocused(true);
    },
    onBlur: (e: React.FocusEvent<HTMLInputElement>) => {
      const maskedValue = maskFn(currentValue, true);
      e.currentTarget.value = maskedValue;
      e.target.value = maskedValue;
      setCurrentValue(maskedValue);

      if (onChange) {
        (onChange as any)(e);
      }

      if (onBlur) {
        (onBlur as any)(e);
      }

      setFocused(false);
    },
    type: 'text',
    inputMode: 'numeric' as const,
    'aria-describedby': mergeIds(originalInputProps['aria-describedby'], labelMaskId),
  };

  let currentMask = maskFn(currentValue);

  // Date mask needs to return the default empty mask when not focused
  if (maskFn === DATE_MASK && !focused) {
    currentMask = maskFn('');
  }

  // SSN mask needs to obfuscate the SSN when not focused
  if (maskFn === SSN_MASK && !focused && currentValue !== '') {
    currentMask = SSN_MASK_OBFUSCATED(currentValue);
    inputProps.value = SSN_MASK_OBFUSCATED(currentValue, true);
  }

  return {
    labelMask: (
      <div className="ds-c-label-mask" id={labelMaskId}>
        <span className={classNames(focused && 'ds-u-visibility--screen-reader')}>
          {currentMask}
        </span>
        <span className={classNames(!focused && 'ds-u-display--none')} aria-hidden="true">
          {currentMask}
        </span>
      </div>
    ),
    inputProps,
  };
}

export default useLabelMask;
