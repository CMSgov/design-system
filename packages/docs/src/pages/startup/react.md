---
title: Using React components
weight: 3
---

The examples below assume you've installed the design system using NPM and have already setup your build system.

<h4 class="ds-h4 ds-u-font-size--base">Default imports</h4>

Individual components can be imported from their individual export file.

```jsx
import Button from '@cmsgov/design-system-core/dist/components/Button/Button';
```

<h4 class="ds-h4 ds-u-font-size--base">Named imports</h4>

Components can also be imported using the shorter syntax below.

<div class="ds-c-alert ds-c-alert--warn ds-u-margin-bottom--2 ds-u-font-size--small">
  <div class="ds-c-alert__body">
    <h3 class="ds-c-alert__heading ds-u-font-size--base">Performance note</h3>
    <p class="ds-c-alert__text">
      This approach may result in a much larger file than you intend. Depending on what module bundler you use, all of the design system's React components may be included in the bundled file even if you didn't specifically import them. This can be avoided by enabling features like [tree shaking in Webpack](https://webpack.js.org/guides/tree-shaking/).
    </p>
  </div>
</div>

```jsx
import { Button } from '@cmsgov/design-system-core';
```

[View an example](https://github.com/CMSgov/design-system/blob/master/examples/react-app/src/scripts/index.js)

<h2 id="examples" class="ds-h2 ds-u-color--primary-darker">Examples</h2>

Additional examples of the design system in use can be viewed on GitHub. These projects demonstrate the various ways you can incorporate the design system into your development process and various use cases.

<a href="https://github.com/CMSgov/design-system/tree/master/examples/" class="ds-c-button">Browse example projects</a>

<h2 id="need-help" class="ds-h2 ds-u-color--primary-darker">Need help or ran into an issue?</h2>

If you're having trouble installing or setting up the design system, or if you think you've found a bug, feel free to [open an issue on GitHub](https://github.com/CMSgov/design-system/issues).
