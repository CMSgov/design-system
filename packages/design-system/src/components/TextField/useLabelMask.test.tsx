import React from 'react';
import useLabelMask, {
  DATE_MASK,
  SSN_MASK,
  SSN_MASK_OBFUSCATED,
  PHONE_MASK,
  ZIP_MASK,
  CURRENCY_MASK,
  MaskFunction,
} from './useLabelMask';
import { render, screen } from '@testing-library/react';
import { act, renderHook } from '@testing-library/react-hooks';
import userEvent from '@testing-library/user-event';

describe('SSN_MASK', () => {
  it('returns just the mask when given no input', () => {
    expect(SSN_MASK('')).toEqual('###-##-####');
  });

  it('masks complete social security numbers', () => {
    expect(SSN_MASK('123-45-6789')).toEqual('123-45-6789');
    expect(SSN_MASK('123 45 6789')).toEqual('123-45-6789');
    expect(SSN_MASK('123456789')).toEqual('123-45-6789');
    expect(SSN_MASK('123.45.6789')).toEqual('123-45-6789');
  });

  it('masks incomplete social security numbers', () => {
    expect(SSN_MASK('123')).toEqual('123-##-####');
    expect(SSN_MASK('1234')).toEqual('123-4#-####');
    expect(SSN_MASK('1234567')).toEqual('123-45-67##');
  });

  it('handles valueOnly parameter', () => {
    expect(SSN_MASK('123', true)).toEqual('123');
    expect(SSN_MASK('12345', true)).toEqual('123-45');
    expect(SSN_MASK('1234567', true)).toEqual('123-45-67');
  });
});

describe('SSN_MASK_OBFUSCATED', () => {
  it('returns just the mask when given no input', () => {
    expect(SSN_MASK_OBFUSCATED('')).toEqual('###-##-####');
  });

  it('masks complete social security numbers', () => {
    expect(SSN_MASK_OBFUSCATED('123-45-6789')).toEqual('***-**-6789');
    expect(SSN_MASK_OBFUSCATED('123 45 6789')).toEqual('***-**-6789');
    expect(SSN_MASK_OBFUSCATED('123456789')).toEqual('***-**-6789');
    expect(SSN_MASK_OBFUSCATED('123.45.6789')).toEqual('***-**-6789');
  });

  it('masks incomplete social security numbers', () => {
    expect(SSN_MASK_OBFUSCATED('123')).toEqual('***-##-####');
    expect(SSN_MASK_OBFUSCATED('1234')).toEqual('***-*#-####');
    expect(SSN_MASK_OBFUSCATED('1234567')).toEqual('***-**-67##');
  });

  it('handles valueOnly parameter', () => {
    expect(SSN_MASK_OBFUSCATED('123', true)).toEqual('***');
    expect(SSN_MASK_OBFUSCATED('12345', true)).toEqual('***-**');
    expect(SSN_MASK_OBFUSCATED('1234567', true)).toEqual('***-**-67');
  });
});

describe('ZIP_MASK', () => {
  it('returns just the mask when given no input', () => {
    expect(ZIP_MASK('')).toEqual('#####');
  });

  it('masks complete zip codes', () => {
    expect(ZIP_MASK('12345')).toEqual('12345');
  });

  it('masks incomplete social zip codes', () => {
    expect(ZIP_MASK('1')).toEqual('1####');
    expect(ZIP_MASK('12')).toEqual('12###');
    expect(ZIP_MASK('123')).toEqual('123##');
    expect(ZIP_MASK('1234')).toEqual('1234#');
  });

  it('handles valueOnly parameter', () => {
    expect(ZIP_MASK('1', true)).toEqual('1');
    expect(ZIP_MASK('12', true)).toEqual('12');
    expect(ZIP_MASK('123', true)).toEqual('123');
  });
});

describe('CURRENCY_MASK', () => {
  it('returns just the mask when given no input', () => {
    expect(CURRENCY_MASK('')).toEqual('$');
  });

  it('returns correctly formatted dollar amounts', () => {
    expect(CURRENCY_MASK('1')).toEqual('$1');
    expect(CURRENCY_MASK('12.479')).toEqual('$12.47');
    expect(CURRENCY_MASK('123456789')).toEqual('$123,456,789');
    expect(CURRENCY_MASK('0.05')).toEqual('$0.05');
    expect(CURRENCY_MASK('$25')).toEqual('$25');
    expect(CURRENCY_MASK('2,300.4')).toEqual('$2,300.40');
    expect(CURRENCY_MASK('-400')).toEqual('-$400');
    expect(CURRENCY_MASK('-428.33')).toEqual('-$428.33');
    expect(CURRENCY_MASK('ABCD$500.22')).toEqual('$');
    expect(CURRENCY_MASK('-2500,0.12')).toEqual('-$25,000.12');
  });
});

describe('PHONE_MASK', () => {
  it('returns just the mask when given no input', () => {
    expect(PHONE_MASK('')).toEqual('###-###-####');
  });

  it('masks complete us phone numbers', () => {
    expect(PHONE_MASK('2225551212')).toEqual('222-555-1212');
    expect(PHONE_MASK('222-555-1212')).toEqual('222-555-1212');
    expect(PHONE_MASK('(222)555-1212')).toEqual('222-555-1212');
    expect(PHONE_MASK('(222) 555-1212')).toEqual('222-555-1212');
    expect(PHONE_MASK('222.555.1212')).toEqual('222-555-1212');
    expect(PHONE_MASK('222 555 1212')).toEqual('222-555-1212');
  });

  it('masks incomplete phone numbers', () => {
    expect(PHONE_MASK('2')).toEqual('2##-###-####');
    expect(PHONE_MASK('222')).toEqual('222-###-####');
    expect(PHONE_MASK('22255')).toEqual('222-55#-####');
    expect(PHONE_MASK('2225551')).toEqual('222-555-1###');
  });

  it('handles valueOnly parameter', () => {
    expect(PHONE_MASK('222', true)).toEqual('222');
    expect(PHONE_MASK('222555', true)).toEqual('222-555');
    expect(PHONE_MASK('2225551212', true)).toEqual('222-555-1212');
  });
});

describe('DATE_MASK', () => {
  it('returns just the mask when given no input', () => {
    expect(DATE_MASK('')).toEqual('MM/DD/YYYY');
  });

  it('masks complete dates', () => {
    expect(DATE_MASK('01-02-2030')).toEqual('01/02/2030');
    expect(DATE_MASK('01 02 2030')).toEqual('01/02/2030');
    expect(DATE_MASK('01.02.2030')).toEqual('01/02/2030');
    expect(DATE_MASK('01/02/2030')).toEqual('01/02/2030');
    expect(DATE_MASK('01022030')).toEqual('01/02/2030');
  });

  it('masks incomplete dates', () => {
    expect(DATE_MASK('01')).toEqual('01/DD/YYYY');
    expect(DATE_MASK('1')).toEqual('01/DD/YYYY');
    expect(DATE_MASK('0102')).toEqual('01/02/YYYY');
    expect(DATE_MASK('012')).toEqual('01/02/YYYY');
    expect(DATE_MASK('0112')).toEqual('01/12/YYYY');
    expect(DATE_MASK('011220')).toEqual('01/12/20YY');
  });

  it('pads zeros', () => {
    expect(DATE_MASK('1-2-2030')).toEqual('01/02/2030');
    expect(DATE_MASK('1-02-2030')).toEqual('01/02/2030');
    expect(DATE_MASK('01-2-2030')).toEqual('01/02/2030');
    expect(DATE_MASK('1')).toEqual('01/DD/YYYY');
  });

  it('handles valueOnly parameter', () => {
    expect(DATE_MASK('01', true)).toEqual('01');
    expect(DATE_MASK('1', true)).toEqual('01');
    expect(DATE_MASK('0102', true)).toEqual('01/02');
    expect(DATE_MASK('012', true)).toEqual('01/02');
    expect(DATE_MASK('0112', true)).toEqual('01/12');
    expect(DATE_MASK('011220', true)).toEqual('01/12/20');
    expect(DATE_MASK('01-2-2030', true)).toEqual('01/02/2030');
  });
});

describe('useLabelMask', () => {
  const defaultInputProps = { type: 'text', value: '', id: 'static-id' };

  function renderInput(props = {}) {
    return render(<input {...props} />);
  }

  function renderUseLabelMask(inputProps = {}, mask: MaskFunction = DATE_MASK) {
    return renderHook(() => useLabelMask(mask, { ...defaultInputProps, ...inputProps }));
  }

  it('returns labelMask and input elements', () => {
    const { result } = renderUseLabelMask();
    const { labelMask, inputProps } = result.current;

    const mask = render(labelMask).container.querySelector('.ds-c-label-mask');

    expect(mask).toBeInTheDocument();
    expect(mask.querySelectorAll('span')).toHaveLength(2);
    expect(mask.textContent).toContain('MM/DD/YYYY');

    const input = renderInput(inputProps).container.querySelector('input');
    expect(input).toBeInTheDocument();
    expect(input).toHaveAttribute('type', 'text');
    expect(input).toHaveAttribute('inputmode', 'numeric');
  });

  it('shows masked value when focused', () => {
    const { result } = renderUseLabelMask({ value: '12' });
    renderInput(result.current.inputProps);
    // Can't really find an alternative that gets rid of the warnings
    // eslint-disable-next-line testing-library/no-unnecessary-act
    act(() => {
      userEvent.click(screen.getByRole('textbox'));
    });
    expect(render(result.current.labelMask).asFragment()).toMatchSnapshot();
  });

  it('shows unfilled mask when not focused', () => {
    const { result } = renderUseLabelMask({ value: '12250001' });
    renderInput(result.current.inputProps);
    // Can't really find an alternative that gets rid of the warnings
    // eslint-disable-next-line testing-library/no-unnecessary-act
    act(() => {
      userEvent.click(screen.getByRole('textbox'));
      userEvent.tab();
    });
    expect(render(result.current.labelMask).asFragment()).toMatchSnapshot();
  });

  it('renders the appropriate dom for a PHONE_MASK', () => {
    const { result } = renderUseLabelMask({}, PHONE_MASK);
    expect(render(result.current.labelMask).asFragment()).toMatchSnapshot();
  });
  it('renders the appropriate dom for a ZIP_MASK', () => {
    const { result } = renderUseLabelMask({}, ZIP_MASK);
    expect(render(result.current.labelMask).asFragment()).toMatchSnapshot();
  });
  it('renders the appropriate dom for a SSN_MASK', () => {
    const { result } = renderUseLabelMask({}, SSN_MASK);
    expect(render(result.current.labelMask).asFragment()).toMatchSnapshot();
  });
  it('renders the appropriate dom for a CURRENCY_MASK', () => {
    const { result } = renderUseLabelMask({}, CURRENCY_MASK);
    expect(render(result.current.labelMask).asFragment()).toMatchSnapshot();
  });
});
