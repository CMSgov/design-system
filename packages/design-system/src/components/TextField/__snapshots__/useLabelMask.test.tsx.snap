// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useLabelMask renders the appropriate dom for a CURRENCY_MASK 1`] = `
<DocumentFragment>
  <div
    class="ds-c-label-mask"
    id="static-id__label-mask"
  >
    <span
      class=""
    >
      $
    </span>
    <span
      aria-hidden="true"
      class="ds-u-display--none"
    >
      $
    </span>
  </div>
</DocumentFragment>
`;

exports[`useLabelMask renders the appropriate dom for a PHONE_MASK 1`] = `
<DocumentFragment>
  <div
    class="ds-c-label-mask"
    id="static-id__label-mask"
  >
    <span
      class=""
    >
      ###-###-####
    </span>
    <span
      aria-hidden="true"
      class="ds-u-display--none"
    >
      ###-###-####
    </span>
  </div>
</DocumentFragment>
`;

exports[`useLabelMask renders the appropriate dom for a SSN_MASK 1`] = `
<DocumentFragment>
  <div
    class="ds-c-label-mask"
    id="static-id__label-mask"
  >
    <span
      class=""
    >
      ###-##-####
    </span>
    <span
      aria-hidden="true"
      class="ds-u-display--none"
    >
      ###-##-####
    </span>
  </div>
</DocumentFragment>
`;

exports[`useLabelMask renders the appropriate dom for a ZIP_MASK 1`] = `
<DocumentFragment>
  <div
    class="ds-c-label-mask"
    id="static-id__label-mask"
  >
    <span
      class=""
    >
      #####
    </span>
    <span
      aria-hidden="true"
      class="ds-u-display--none"
    >
      #####
    </span>
  </div>
</DocumentFragment>
`;

exports[`useLabelMask shows masked value when focused 1`] = `
<DocumentFragment>
  <div
    class="ds-c-label-mask"
    id="static-id__label-mask"
  >
    <span
      class="ds-u-visibility--screen-reader"
    >
      12/DD/YYYY
    </span>
    <span
      aria-hidden="true"
      class=""
    >
      12/DD/YYYY
    </span>
  </div>
</DocumentFragment>
`;

exports[`useLabelMask shows unfilled mask when not focused 1`] = `
<DocumentFragment>
  <div
    class="ds-c-label-mask"
    id="static-id__label-mask"
  >
    <span
      class=""
    >
      MM/DD/YYYY
    </span>
    <span
      aria-hidden="true"
      class="ds-u-display--none"
    >
      MM/DD/YYYY
    </span>
  </div>
</DocumentFragment>
`;
